oxd-server/src/test/java/org/xdi/oxd/server/GetTokensByCodeTest.java

```diff
@@ -72,6 +74,84 @@ public static GetTokensByCodeResponse tokenByCode(ClientInterface client, Regist
         return resp;
     }
 
+    @Parameters({"host", "opHost", "redirectUrl", "userId", "userSecret"})
+    @Test
+    public void testWithInvalidAuthorizationCode(String host, String opHost, String redirectUrl, String userId, String userSecret) throws Exception {
+
+        final ClientInterface client = Tester.newClient(host);
+
+        final RegisterSiteResponse site = RegisterSiteTest.registerSite(client, opHost, redirectUrl);
+
+        final String state = CoreUtils.secureRandomString();
+
+        final String authorizationStr = Tester.getAuthorization();
+
+        final String code = codeRequest(client, site.getOxdId(), userId, userSecret, state, CoreUtils.secureRandomString());
```

Comment:
redundant line, we don't need valid code if we are going to make request with invalid one.

Reply:
The line is not redundant. It actually registers the state on the server. If we don't add this line, then it will throw state not registered or sort of that error and not bad code error.